packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/EditForm/useEditForm.js

```diff
@@ -0,0 +1,57 @@
+import { useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../../context/cart';
+
+export const useEditForm = props => {
+    const {
+        afterSubmit,
+        mutations: { setShippingInformationMutation },
+        shippingData
+    } = props;
+
+    const [{ cartId }] = useCartContext();
+    const [setShippingInformation, { called, loading }] = useMutation(
+        setShippingInformationMutation
+    );
+
+    const { country, region } = shippingData;
+    const { code: countryCode } = country;
+    const { code: regionCode } = region;
+
+    const initialValues = {
+        ...shippingData,
+        country: countryCode,
+        region: regionCode
+    };
+
+    const isUpdate = !!shippingData.email;
```

Comment:
A similar note here as earlier -- is presence of email enough to know if this is an update or an initial input?

Reply:
Yes, currently in the cart/checkout experience, you can't set an email any other way.